SQF/dayz_code/compile/dze_buildChecks.sqf

```diff
@@ -73,5 +73,22 @@ if (_toolCheck) then {
 	_canBuild = [_item, _require, _classname] call dze_requiredItemsCheck;
 };
 
+_text = getText (configFile >> 'CfgMagazines' >> _item >> 'displayName');
+
+if (((count DZE_SafeZoneNoBuildItems) > 0) && {_classname in DZE_SafeZoneNoBuildItems}) then {
+	{
+		if ((player distance (_x select 0)) <  DZE_SafeZoneNoBuildDistance) exitWith { _canBuild = false; };
+	} forEach DZE_safeZonePosArray;
+};
+
+if !(_canBuild) exitWith { dayz_actionInProgress = false; format [localize "STR_EPOCH_PLAYER_166",_text,DZE_SafeZoneNoBuildDistance] call dayz_rollingMessages; [false, _isPole]; };
+
+if ((count DZE_NoBuildNear) > 0) then {
+	_near = (nearestObjects [_pos,DZE_NoBuildNear,DZE_NoBuildNearDistance]);
+	if ((count _near) > 0) then { _canBuild = false; };
+};
+
+if !(_canBuild) exitWith { dayz_actionInProgress = false; format [localize "STR_EPOCH_PLAYER_167",_text,DZE_NoBuildNearDistance,typeOf (_near select 0)] call dayz_rollingMessages; [false, _isPole]; };
+
```

Comment:
This [whole block](https://github.com/oiad/DayZ-Epoch/blob/8c0c8732a53a2e3ab52c3469a948173f22a35332/SQF/dayz_code/compile/dze_buildChecks.sqf#L76-L91) should be moved above the (_toolCheck), otherwise it is exiting with the wrong message when the has required tools check returns false.